File: Data Structures & Algorithms/missing-number/submission-2.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        range_nums = []

        for i in range(len(nums)+1):
            range_nums.append(i)

        res = 0
        for num in range_nums:
            if num not in nums:
                return num
        
        return res

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-2.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer, Integer> preMap = new HashMap<>();
        preMap.put(0, 1);
        int curSum = 0, res = 0;

        for(int num: nums) {
            curSum += num;
            int diff = curSum - k;

            res += preMap.getOrDefault(diff, 0);
            preMap.put(curSum, preMap.getOrDefault(curSum, 0) + 1);
        }

        return res;
    }
}